modules/transactions/src/main/java/org/apache/ignite/internal/tx/views/TransactionsViewProvider.java

```diff
@@ -93,27 +86,20 @@ public SystemView<?> get() {
     static class TxInfoDataSource implements Iterable<TxInfo> {
         private final UUID localNodeId;
 
-        private final Iterable<UUID> roTxIds;
-
         private final Map<UUID, TxStateMeta> rwTxStates;
 
-        TxInfoDataSource(UUID localNodeId, Iterable<UUID> roTxIds, Map<UUID, TxStateMeta> rwTxStates) {
+        TxInfoDataSource(UUID localNodeId, Map<UUID, TxStateMeta> rwTxStates) {
             this.localNodeId = localNodeId;
-            this.roTxIds = roTxIds;
             this.rwTxStates = rwTxStates;
         }
 
         @Override
         public Iterator<TxInfo> iterator() {
-            return CollectionUtils.concat(
-                    new TransformingIterator<>(roTxIds.iterator(), TxInfo::readOnly),
-                    rwTxStates.entrySet().stream()
+            return rwTxStates.entrySet().stream()
                             .filter(e -> localNodeId.equals(e.getValue().txCoordinatorId())
-                                    && e.getValue().tx() != null && !e.getValue().tx().isReadOnly()
-                                    && !isFinalState(e.getValue().txState()))
-                            .map(e -> TxInfo.readWrite(e.getKey(), e.getValue().txState()))
-                            .iterator()
-            );
+                                    && e.getValue().tx() != null && !e.getValue().tx().isFinishingOrFinished())
```

Comment:
I think, previous condition `!isFinalState(e.getValue().txState())` was more correct than new one `!e.getValue().tx().isFinishingOrFinished()`. The latter excludes transactions in FINISHING state, which is quite important.

Reply:
Thanks, this check has been changed because RO tx currently does not change state after completion in the state map (it always has state PENDING).

I reworked this code a bit to do separate checks for RW and RO tx, and added a comment about the RO state.